### PR TITLE
 [FIX] product: add domain on ptav filter

### PR DESCRIPTION
**Pull request by kcv-odoo**
### [FIX] product: add domain on ptav filter
Steps:
- Install sale.
- If user using odoo for 3-4 for years then
user probably have more then 200 ptavs in db..

Issue:
- It'll display limit warning instead of filter
in most of cases.

Cause:
- Have larger number of ptavs.

Fix:
-  Add domain to only search active ptav and has
related active product.

opw-3827751

### CODE REVIEW

**Comment by kcv-odoo** [comment]
Was not easily able to reproduce even with steps :(

File: addons/product/views/product_views.xml
Comment (Feyensv) [inline addons/product/views/product_views.xml:622]: Isn't it on the ptav filter that you want to remove the counters ?

**Review by Feyensv — COMMENTED** [review]
`web` part has to be approved by the framework team(s) first.

File: addons/product/views/product_views.xml
Comment (kcv-odoo) [reply addons/product/views/product_views.xml:622]: We only need to remove counter from `category` type section not `filter` type section [see](https://github.com/odoo/odoo/blob/17.0/addons/web/static/src/search/search_arch_parser.js#L342)
Having counter on filter type does not cause any issue I guess but we can remove it if it is not very useful :thinking: 
[Difference b/t category and filter type](https://github.com/odoo/odoo/blob/17.0/addons/web/static/src/search/search_arch_parser.js#L301)

**Comment by kcv-odoo** [comment]
Hello @odoo/rd-framework-py :wave: 
Can you review first commit?

Thank you!

**Comment by kmagusiak** [comment]
> * ch active

Seems ok, the error will appear more often because when `not expand` only some of the values were added.
@rco-odoo Can you confirm?

**Comment by mctm-odoo** [comment]
Hello @kcv-odoo I see you are releasing this fix for v. 17 I have a case that's exactly like this on v. 17.2. Will this fix be released for other versions as well?

**Comment by kcv-odoo** [comment]
Hello @mctm-odoo 
Yes this fix will be released for other versions once @odoo/rd-framework-py review and approve these changes

**Comment by Polymorphe57** [comment]
A fix <https://github.com/odoo/odoo/pull/172963> will be merged in 15.0 and after.  So only your second commit will be useful here.

**Comment by Feyensv** [comment]
@kcv-odoo can you drop your first commit then so we can merge this PR ? (you'll have to wait until both commits are merged in 17 before telling the customer the issue is fixed).

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**
